spring-cloud-gcp-data-firestore/src/main/java/org/springframework/cloud/gcp/data/firestore/FirestoreTemplate.java

```diff
@@ -124,34 +136,76 @@ public FirestoreTemplate(FirestoreStub firestore, String parent) {
 						.map(document -> PublicClassMapper.convertToCustomClass(document, clazz)));
 	}
 
+	public <T> Mono<Long> count(Class<T> entityClass) {
+		return findAllDocuments(entityClass, ID_PROJECTION).count();
```

Comment:
This is going to have a performance penalty relative to expected `count()` performance, so we need to document it very well.